bokehjs/src/coffee/models/ranges/factor_range.ts

```diff
@@ -142,14 +142,19 @@ export class FactorRange extends Range {
   }
 
   protected _lookup(x: any): number {
-    if (x.length == 1)
-      return (this._mapping as L1Mapping)[x[0]].value
-    else if (x.length == 2)
-      return (this._mapping as L2Mapping)[x[0]].mapping[x[1]].value
-    else if (x.length == 3)
-      return (this._mapping as L3Mapping)[x[0]].mapping[x[1]].mapping[x[2]].value
-    else
-      return undefined as never
+    try {
+      if (x.length == 1)
+        return (this._mapping as L1Mapping)[x[0]].value
+      else if (x.length == 2)
+        return (this._mapping as L2Mapping)[x[0]].mapping[x[1]].value
+      else if (x.length == 3)
+        return (this._mapping as L3Mapping)[x[0]].mapping[x[1]].mapping[x[2]].value
+      else
+        return undefined as never
+    }
+    catch (e) {
```

Comment:
At minimum this needs a comment regarding what this code guards against. However, this code just asks for causing problems in future. I would handle missing factors explicitly at each step.